pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -236,9 +257,19 @@ public String segmentCommit(@QueryParam(SegmentCompletionProtocol.PARAM_INSTANCE
     SegmentCompletionProtocol.Response response = segmentCompletionManager.segmentCommitStart(requestParams);
     if (response.equals(SegmentCompletionProtocol.RESP_COMMIT_CONTINUE)) {
       // Get the segment and put it in the right place.
-      boolean success = uploadSegment(multiPart, instanceId, segmentName, false) != null;
+      URI segmentURI = uploadSegment(multiPart, instanceId, segmentName, false);
```

Comment:
Please correct me if I am wrong, but it seems that we get the segment to a tmp dir locally, store it in the file system, download it again, and retrieve the metadata. We should be retrieving metadata before we upload,.

Reply:
Done. Now in uploadSegment(), the metadata will be extracted directly for the uploaded file if it is not a split commit.